### PR TITLE
Issue #26 - Execute all Executions in parallel

### PR DESCRIPTION
**Pull request by pborges**

### CODE REVIEW

File: smarthome.go [hunk 1]
```diff
@@ -67,15 +69,66 @@ func (s *SmartHome) Handle() http.HandlerFunc {
...
+		for _, cmd := range trait.TraitCommands() {
+			// for the right command
+			if ex.Command == cmd.Name() {
+				// and execute
+				ctx := Context{Target: dev}
+				if err := cmd.Execute(ctx, ex.Params); err == nil {
+					response.Status = proto.CommandStatusSuccess
+					response.ErrorCode = ""
+
+					// States are optional in this response
```
Comment (localleon) [inline smarthome.go:101]: Does the response time dramaticly increase if you leave this in?

File: smarthome.go [hunk 2]
```diff
@@ -84,55 +137,43 @@ func (s *SmartHome) handleExecuteIntent(agentUserId string, req proto.ExecReques
 		}
 	}
 	responseBody := proto.ExecResponse{}
-	// for each command
+	responses := make(chan proto.CommandResponse)
+	resCount := 0
```
Comment (localleon) [inline smarthome.go]: Maybe just `var resCount int`, could be a little bit clearer and would produce the same results ?

File: smarthome.go [hunk 3]
```diff
@@ -176,7 +217,9 @@ func (s *SmartHome) encodeDeviceForSyncResponse(dev Device) proto.Device {
 }
 
 func (s *SmartHome) handleSyncIntent(agentUserId string) proto.SyncResponse {
-	log.Printf("[%s] SYNC\n", agentUserId)
+	if s.Log != nil {
+		s.Log.Printf("[%s] SYNC REQUEST\n", agentUserId)
+	}
 
 	devices := make([]proto.Device, 0)
```
Comment (localleon) [inline smarthome.go]: You could move this directly into the struct. 
Don't know which one's better for readability

```
	response := proto.SyncResponse{
		AgentUserId: agentUserId,
		Devices:     make([]proto.Device, 0),
	}
```

**Review by localleon — COMMENTED** [review]
Some minor comments. Looks good so far. Performs as expected in real world scenarios.
+1 for the Comments

File: smarthome.go (hunk 1 above)
Comment (pborges) [reply smarthome.go:101]: unknown, it depends on what the user has defined in the state handlers, for example with insteon it means doing a HTTP request, and their API responds slowly

**Pull request merged**